contracts/src/core/AlignedLayerServiceManager.sol

```diff
@@ -98,30 +105,35 @@ contract AlignedLayerServiceManager is
             abi.encodePacked(batchMerkleRoot, senderAddress)
         );
 
-        /* CHECKING SIGNATURES & WHETHER THRESHOLD IS MET OR NOT */
+        BatchState storage currentBatch = batchesState[batchIdentifierHash];
 
         // Note: This is a hacky solidity way to see that the element exists
         // Value 0 would mean that the task is in block 0 so this can't happen.
-        if (batchesState[batchIdentifierHash].taskCreatedBlock == 0) {
+        if (currentBatch.taskCreatedBlock == 0) {
             revert BatchDoesNotExist(batchIdentifierHash);
         }
 
         // Check task hasn't been responsed yet
-        if (batchesState[batchIdentifierHash].responded) {
+        if (currentBatch.responded) {
             revert BatchAlreadyResponded(batchIdentifierHash);
         }
+        currentBatch.responded = true; 
 
-        if (batchersBalances[senderAddress] == 0) {
-            revert BatcherHasNoBalance(senderAddress);
+        // Check that batcher has enough funds to fund response
+        if (batchersBalances[senderAddress] < currentBatch.maxFeeAllowedToRespond) {
+            revert InsufficientFunds(
+                senderAddress,
+                currentBatch.maxFeeAllowedToRespond,
+                batchersBalances[senderAddress]
+            );
         }
```

Comment:
This is fine for now and better than what we had, but a batcher may have insufficient funds to pay for a second or third pending batch

Reply:
Yes, but for now this can't be checked on createNewTask.
Also, Batcher may have N pending batches that will never be accepted, and he should not need to have funds for these batches.